x-pack/plugin/data-frame/src/main/java/org/elasticsearch/xpack/dataframe/action/TransportDeleteDataFrameTransformAction.java

```diff
@@ -56,19 +64,34 @@ protected AcknowledgedResponse read(StreamInput in) throws IOException {
 
     @Override
     protected void masterOperation(Task task, Request request, ClusterState state,
-                                   ActionListener<AcknowledgedResponse> listener) throws Exception {
-        PersistentTasksCustomMetaData pTasksMeta = state.getMetaData().custom(PersistentTasksCustomMetaData.TYPE);
-        if (pTasksMeta != null && pTasksMeta.getTask(request.getId()) != null) {
+                                   ActionListener<AcknowledgedResponse> listener) {
+        final PersistentTasksCustomMetaData pTasksMeta = state.getMetaData().custom(PersistentTasksCustomMetaData.TYPE);
+        if (pTasksMeta != null && pTasksMeta.getTask(request.getId()) != null && request.isForce() == false) {
             listener.onFailure(new ElasticsearchStatusException("Cannot delete data frame [" + request.getId() +
                     "] as the task is running. Stop the task first", RestStatus.CONFLICT));
         } else {
-            // Task is not running, delete the configuration document
-            transformsConfigManager.deleteTransform(request.getId(), ActionListener.wrap(
-                    r -> {
-                        auditor.info(request.getId(), "Deleted data frame transform.");
-                        listener.onResponse(new AcknowledgedResponse(r));
-                    },
-                    listener::onFailure));
+            ActionListener<Void> stopTransformActionListener = ActionListener.wrap(
```

Comment:
[optional] IMO more readable code could be achieved by:
```
CheckedRunnable deleteTransform = () -> transformsConfigManager.deleteTransform(..., ...);
if (pTasksMeta != null && pTasksMeta.getTask(request.getId()) != null) {
  executeAsyncWithOrigin(
    ...,
    ActionListener.wrap(r -> deleteTransform.run(), listener::onFailure));
} else {
    deleteTransform.run();
}
```  

I see 2 advantages here:
1. variable "deleteTransform" clearly describes what it does and is not tied to some other action's listener.
2. no need to pass unused parameters to it
WDYT?

Reply:
I prefer to try and keep callbacks uniformed. Our callback stacks can get very complicated and to me its just easier to reason about if they are `ActionListener` + and always within the flow (if possible)